src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +30,50 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        assignDefaultValues();
+        splitAddress(address);
+    }
+    
+    /**
+     * Assigns default values to Block, Street, Unit and Postal Code should they never be initialized.
+     */
+    private void assignDefaultValues() {
```

Comment:
please think through if this is necessary? Java anyway provides with a default constructor that assigns null values to the members.